src/OpenSage.Game/Data/Wav/WavFile.cs

```diff
@@ -65,51 +135,118 @@ public sealed class WavFile
+            using (var writer = new BinaryWriter(memoryStream))
             {
-                throw new InvalidDataException("Invalid .wav file!");
+                for (int i = 0; i < numBlocks; i++)
+                {
+                    int sample = reader.ReadInt16();
+                    int stepTableIndex = reader.ReadByte();
+                    byte reserved = reader.ReadByte(); // unused, commonly 0
+                    int step = imaStepTable[stepTableIndex];
+                    writer.Write((Int16) sample);
+                    for (int j = 0; j < (samplesPerBlock - 1) / 2; j++)
+                    {
+                        byte packed = reader.ReadByte();
+                        byte nibble_low = (byte) (packed & 0xF);
+                        byte nibble_high = (byte) (packed >> 4);
+                        foreach (var nibble in new byte[] { nibble_low , nibble_high })
+                        {
+                            stepTableIndex += imaIndexTable[nibble];
+                            if (stepTableIndex > 88) stepTableIndex = 88;
+                            if (stepTableIndex < 0) stepTableIndex = 0;
+                            byte sign = (byte) (nibble & 8);
+                            byte delta = (byte) (nibble & 7);
+                            int diff = step >> 3;
+                            if ((delta & 4) != 0) diff += step;
+                            if ((delta & 2) != 0) diff += (step >> 1);
+                            if ((delta & 1) != 0) diff += (step >> 2);
+                            if (sign != 0) sample -= diff;
+                            else sample += diff;
+                            step = imaStepTable[stepTableIndex];
+                            if (sample > 0xFFFF) sample = 0xFFFF;
+                            if (sample < -0xFFFF) sample = -0xFFFF;
+                            writer.Write((Int16) sample);
+                        }
+                    }
+                }
             }
+            return buffer;
+        }
 
-            if(_format.AudioFormat!=1)
-            {
-                throw new NotSupportedException("Invalid .wav compression!");
-            }
+        public void Parse(BinaryReader reader)
+        {
+            _header = RiffHeader.Parse(reader);
+            _format = WaveFormat.Parse(reader);
 
-            //probably a compressed format
-            if (_format.SubChunkSize > 16)
+            // Every format except the original Microsoft format must have a fact chunk
+            if ((AudioFormatType) _format.AudioFormat != AudioFormatType.Microsoft)
             {
-                reader.BaseStream.Seek(_format.SubChunkSize - 16, SeekOrigin.Current);
+                _fact = WaveFact.Parse(reader);
             }
-                
-            _data = WaveData.Parse(reader);
 
-            if (new string(_data.SubChunkID) != "data")
+            _data = WaveData.Parse(reader);
+            switch ((AudioFormatType) _format.AudioFormat)
```

Comment:
As i said this would not be necessary if you created a baseclass for the decoders and abstracted this part away.